caddytls/filestorage_test.go

```diff
@@ -0,0 +1,6 @@
+package caddytls
+
+// *********************************** NOTE ********************************
+// Due to circular package dependencies with the storagetest sub package and
+// the fact that we want to use that harness to test file storage, the tests
+// for file storage are done in the storagetest package.
```

Comment:
This comment doesn't relate to the fact that I can't mock the filesystem or anything. It's more related to the fact that I built a test harness for use by any other storage provider implementors to use, but filestorage_test.go can't use it because it would cause a circular dependency. Therefore, the filestorage impl is actually tested in the `storagetest` package.